motr/examples/example1.c

```diff
@@ -0,0 +1,465 @@
+        struct m0_bufvec   attr;
+
+	uint64_t           last_offset = 0;
+
+	M0_SET0(&obj);
+	instance = container->co_realm.re_instance;
+	m0_obj_init(&obj, &container->co_realm, &obj_id,
+		    m0_client_layout_id(instance));
+
+	rc = object_open(&obj);
+	if (rc != 0) {
+		printf("Failed to open object: rc=%d\n", rc);
+		return rc;
+	}
+
+	/*
+	 * alloc & prepare ext, data and attr. We will write 4k * 2.
+	 */
+	rc = alloc_vecs(&ext, &data, &attr, 2, 4096);
+	if (rc != 0) {
+		printf("Failed to alloc ext & data & attr: %d\n", rc);
+		goto out;
+	}
+	prepare_ext_vecs(&ext, &data, &attr, 2, 4096, &last_offset, 'A');
+
+	/* Start to write data to object */
+	rc = write_data_to_object(&obj, &ext, &data, &attr);
+	cleanup_vecs(&ext, &data, &attr);
+
+out:
+	/* Similar to close() */
+	m0_entity_fini(&obj.ob_entity);
+
+	printf("Object write: %d\n", rc);
+	return rc;
+}
+
+static int read_data_from_object(struct m0_obj      *obj,
+				 struct m0_indexvec *ext,
+				 struct m0_bufvec   *data,
+				 struct m0_bufvec   *attr)
+{
+	int          rc;
+	struct m0_op *ops[1] = { NULL };
+
+	/* Create the write request */
+	m0_obj_op(obj, M0_OC_READ, ext, data, attr, 0, 0, &ops[0]);
+	if (ops[0] == NULL) {
+		printf("Failed to init a read op\n");
+		return -EINVAL;
+	}
+
+	/* Launch the write request*/
+	m0_op_launch(ops, 1);
+
+	/* wait */
+	rc = m0_op_wait(ops[0],
+			M0_BITS(M0_OS_FAILED,
+				M0_OS_STABLE),
+			M0_TIME_NEVER);
+	rc = rc ? : ops[0]->op_sm.sm_rc;
+
+	/* fini and release the ops */
+	m0_op_fini(ops[0]);
+	m0_op_free(ops[0]);
+	printf("Object read result: %d\n", rc);
+	return rc;
+}
+
+static void verify_show_data(struct m0_bufvec *data,
+			     char c)
+{
+	int i, j;
+
+	for (i = 0; i < data->ov_vec.v_nr; ++i) {
+		printf("Block %6d:\n", i);
+		printf("%.*s", (int)data->ov_vec.v_count[i],
+			       (char *)data->ov_buf[i]);
+		printf("\n");
+		for (j = 0; j < data->ov_vec.v_count[i]; j++)
+			if (((char*) data->ov_buf[i])[j] != c) {
+				printf("verification failed at: %d:%d\n"
+				       "Expected %c result %c\n",
+					i, j, c, ((char*)data->ov_buf[i])[j]);
+			}
+	}
+}
+
+static int object_read(struct m0_container *container)
+{
+	struct m0_obj      obj;
+	struct m0_client  *instance;
+	int                rc;
+
+        struct m0_indexvec ext;
+        struct m0_bufvec   data;
+        struct m0_bufvec   attr;
+
+	uint64_t           last_offset = 0;
+
+	M0_SET0(&obj);
+	instance = container->co_realm.re_instance;
+	m0_obj_init(&obj, &container->co_realm, &obj_id,
+		    m0_client_layout_id(instance));
+
+	rc = object_open(&obj);
+	if (rc != 0) {
+		printf("Failed to open object: rc=%d\n", rc);
+		return rc;
+	}
+
+	/*
+	 * alloc & prepare ext, data and attr. We will write 4k * 2.
+	 */
+	rc = alloc_vecs(&ext, &data, &attr, 2, 4096);
+	if (rc != 0) {
+		printf("Failed to alloc ext & data & attr: %d\n", rc);
+		goto out;
+	}
+	prepare_ext_vecs(&ext, &data, &attr, 2, 4096, &last_offset, '\0');
+
+	/* Start to write data to object */
```

Comment:
Start to read data from object ?

Reply:
Thank you, this is a typo.
I will fix this typo and other typos.